k4MarlinWrapper/k4MarlinWrapper/util/EDM4hep2LcioUtils.h

```diff
@@ -0,0 +1,60 @@
+#ifndef EDM4HEP2LCIOUTILS_H
+#define EDM4HEP2LCIOUTILS_H
+
+namespace k4MW::util {
+
+// Reorder parameters based on dependencies between collections
+// This avoids (partly) the need for FillMissingCollections
+template <typename T>
+void optimizeOrderParams(
+  T& params)
+{
+
+  if (params.size() % 3 == 0) {
+    // Get only to the typesin the i%3==0 index
+    std::vector<std::string> params_view;
+    for (int i=0; i < params.size(); i=i+3) {
+      params_view.emplace_back(params[i]);
+    }
+
+    // Can't run in parallel
+    // Dependency should be converted before the target, so swap them (and their names) if out of order.
+    // search for the target and its dependency string, get indexes,
+    // if target before dependency swap all related elements. Multiply indexes by 3 to get correct index from view.
+    auto swap_if_before = [&params_view, &params](const std::string& target, const std::string& dependency)
+    {
+      auto target_found_it = std::find(params_view.begin(), params_view.end(), target);
```

Comment:
you need to find all the targets, as there can be more than one

Reply:
I changed the algorithm to take into account multiple repeated types, so it now searches for all targets, dependencies, etc.
I added a unit test with that case.